node/node_test.go

```diff
@@ -20,19 +20,23 @@ func TestNodeStartStop(t *testing.T) {
 	n.Start()
 	t.Logf("Started node %v", n.sw.NodeInfo())
 
-	// Wait a bit to initialize
-	// TODO remove time.Sleep(), make asynchronous.
-	time.Sleep(time.Second * 2)
+	ticker := time.NewTicker(10 * time.Millisecond)
+	select {
+	case <-ticker.C:
+		if n.IsRunning() {
+			return
```

Comment:
doesnt this abort the test? maybe you mean `break`?

Reply:
Also `IsRunning` should be true right away. Maybe we should wait for a block to be produced?